docs/explanations/default-options.rst

```diff
@@ -0,0 +1,63 @@
+.. SPDX-FileCopyrightText: 2023 The meson-python developers
+..
+.. SPDX-License-Identifier: MIT
+
+.. _explanations-default-options:
+
+*********************
+Default build options
+*********************
+
+Meson offers many `built-in options <https://mesonbuild.com/Builtin-options.html>`__,
+and in the vast majority of cases those have good defaults. There are a couple
+of cases however where ``meson-python`` either is forced or chooses to override
+those with its own defaults. To view what those are for the version of
+``meson-python`` you have installed, look at the *User defined options* section
+of the output during the configure stage of the build (e.g., by running
+``python -m build --wheel``). This will look something like:
+
+.. code-block:: text
+
+    User defined options
+      Native files     : /home/username/code/project/.mesonpy-native-file.ini
+      debug            : false
+      optimization     : 2
+      prefix           : /home/username/mambaforge/envs/project-dev
+      python.platlibdir: /home/username/mambaforge/envs/project-dev/lib/python3.10/site-packages
+      python.purelibdir: /home/username/mambaforge/envs/project-dev/lib/python3.10/site-packages
+      b_ndebug         : if-release
+
+Let's go through each option and why they are used:
+
+- meson-python uses a native file, named ``.mesonpy-native-file.ini`` in order
```

Comment:
The native file will not named like this anymore in the next meson-python release. It will be in the build directory and names `meson-python-native-file.ini`. One thing that is not obvious is that multiple `--native-file` options can be passed and Meson will merge the contents. Therefore, if the user needs to pass any other configuration via a native file, they can add another `--native-file` argument, as long as they do not try to set the path for the `python` executable. I think it would be useful to mention this in the documentation.

Reply:
Agreed, thanks. Will update and add that info.

Reply:
added now